src/test/java/com/mongodb/kafka/connect/source/producer/SchemaAndValueProducerTest.java

```diff
@@ -160,7 +160,7 @@ void testInferSchemaAndValueProducer() {
                 SchemaBuilder.array(
                         SchemaBuilder.struct()
                             .field("a", Schema.OPTIONAL_INT32_SCHEMA)
-                            .name("arrayComplex_a")
```

Comment:
Sigh - the test asserter was also incorrect and didn't recurse correctly else we would have found this error out.